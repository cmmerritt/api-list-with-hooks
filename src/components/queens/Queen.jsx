import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Queen = ({ name, image, quote, id }) => (
  <figure>
    <Link to={`/queen/${id}`}>
      <img src={image} alt={name} height="200px"/>
      <figcaption>
        <p>{name}</p>
        <p>&quot;{quote}&quot;</p>
      </figcaption>
    </Link>
  </figure>
);

Queen.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default Queen;
