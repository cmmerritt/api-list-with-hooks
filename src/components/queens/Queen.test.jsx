import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Queen from './Queen';

describe('Queen component', () => {
  afterEach(() => cleanup());
  it('renders Queen', () => {
    const { asFragment } = render(
      <Queen 
        name="RuPaul"
        image="rupaul.jpg"
        quote="Shante, you stay."
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
