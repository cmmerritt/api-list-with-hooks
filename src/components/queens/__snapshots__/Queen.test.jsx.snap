// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queen component renders Queen 1`] = `
<DocumentFragment>
  <figure>
    <a
      href="/queen/undefined"
    >
      <img
        alt="RuPaul"
        height="200px"
        src="rupaul.jpg"
      />
      <figcaption>
        <p>
          RuPaul
        </p>
        <p>
          "Shante, you stay."
        </p>
      </figcaption>
    </a>
  </figure>
</DocumentFragment>
`;
